File: face-detection/google-cloud-vision/GoogleCloudVisionFaceDetection.kt
// "input" is a File object pointing to the input image file.

val content = ByteString.copyFrom(input.readBytes())

val image = Image.newBuilder()
    .setContent(content)
    .build()
val feature = Feature.newBuilder()
    .setType(Feature.Type.FACE_DETECTION)
    .build()
val request = AnnotateImageRequest.newBuilder()
    .addFeatures(feature)
    .setImage(image)
    .build()

val requests = listOf(request)

val client = ImageAnnotatorClient.create()
val batchResponse = client.batchAnnotateImages(requests)
val response = batchResponse.responsesList.single()

val results = response.faceAnnotationsList
val result = results.single()

File: face-detection/google-cloud-vision/google-cloud-vision-face-detection.js
// "input" is a string containg the path to an image file.

const vision = require('@google-cloud/vision')

const client = new vision.ImageAnnotatorClient()

client
    .faceDetection(input)
    .then(res => {
        const faceDetection = res[0].faceAnnotations
    })
    .catch(err => {
    })
